src/lib.rs

```diff
@@ -29,15 +30,59 @@ mod dlmalloc;
 #[cfg(all(feature = "global", not(test)))]
 mod global;
 
+/// A platform interface
+pub trait System: Send {
+    /// Allocates a memory region of `size` bytes
+    unsafe fn alloc(size: usize) -> (*mut u8, usize, u32);
+
+    /// Remaps a memory region
+    unsafe fn remap(ptr: *mut u8, oldsize: usize, newsize: usize, can_move: bool) -> *mut u8;
+
+    /// Frees a part of a memory region
+    unsafe fn free_part(ptr: *mut u8, oldsize: usize, newsize: usize) -> bool;
+
+    /// Frees an entire memory region
+    unsafe fn free(ptr: *mut u8, size: usize) -> bool;
+
+    /// Indicates if the platform can release a part of memory
+    fn can_release_part(flags: u32) -> bool;
+
+    /// Allocates a memory region of zeros
+    fn allocates_zeros() -> bool;
+
+    /// Returns the page size
+    fn page_size() -> usize;
+}
+
+/// A platform interface for platforms that support the global lock
+#[cfg(feature = "global")]
+pub trait GlobalSystem: System {
```

Comment:
This seems like it's not particularly necessary for this PR? Could this perhaps exempt unknown platforms from a global allocator if locking is unknown?

Reply:
Indeed. I've reverted the change